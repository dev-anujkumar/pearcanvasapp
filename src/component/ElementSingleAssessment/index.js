import {ElementSingleAssessment} from './ElementSingleAssessment.jsx';
export default ElementSingleAssessment;