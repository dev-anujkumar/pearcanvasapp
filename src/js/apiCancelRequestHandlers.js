/**
 * This module consists of a handler function for cancelling axios API request in case of multiple API requests
 */
import axios from 'axios';
import config from '../config/config';

const axiosRequestCreator = () => {
    let apiCall;
    return url => {
        if (apiCall) {
            apiCall.cancel("Only one request allowed at a time.");
        }
        apiCall = axios.CancelToken.source();
        return axios.get(url, {
            cancelToken: apiCall.token,
            headers:{
            PearsonSSOSession: config.ssoToken,
            'myCloudProxySession': config.myCloudProxySession            
        }});
    };
};
export const axiosGetAPI = axiosRequestCreator();