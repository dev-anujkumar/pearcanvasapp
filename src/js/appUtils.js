import cypressConfig from '../config/cypressConfig';
import { sendDataToIframe} from '../constants/utility';

export const modifyObjKeys = (obj, newObj) => {
    Object.keys(obj).forEach(function(key) {
      delete obj[key];
    });      
    Object.keys(newObj).forEach(function(key) {
      obj[key] = newObj[key];
    });
    Object.keys(cypressConfig).forEach(function(key) {
        obj[key] = cypressConfig[key];
    });
    console.log("Get MongoDB taskdef config>> ", obj);
    if(process.env.NODE_ENV === 'development'){
        obj.REACT_APP_API_URL = cypressConfig.CYPRESS_API_ENDPOINT;
        obj.JAVA_API_URL = cypressConfig.CYPRESS_TOC_JAVA_ENDPOINT;
    }
}

export const timeSince = (countTimer) => {
    let count;
    const intervals = [
        { label: 'year', seconds: 31536000 },
        { label: 'month', seconds: 2592000 },
        { label: 'day', seconds: 86400 },
        { label: 'hour', seconds: 3600 },
        { label: 'minute', seconds: 60 },
        { label: 'second', seconds: 0 }
    ];
    let seconds = Math.floor((new Date().getTime() - countTimer) / 1000);
    let interval = intervals.find(i => i.seconds <= seconds);
    if (interval && interval.label != 'second') {
        count = Math.floor(seconds / interval.seconds);
        sendDataToIframe({ 'type': 'slateRefreshStatus', 'message': {slateRefreshStatus : `Refreshed, ${count} ${interval.label == 'second' ? '' : interval.label} ago`} });
    }        
}

export const removeWirisOverlay =() =>{
    let targetNode = document.querySelector('body');
        // Options for the observer (which mutations to observe)		
        var config = { attributes: true };
        // Callback function to execute when mutations are observed		
        let callbackOb = function (mutationsList, observercb) {
            for (var mutation of mutationsList) {
                if (mutation.type === 'attributes') {
                    let wirisNodes = document.getElementsByClassName('wrs_modal_dialogContainer');
                    let wirisNodeLength = wirisNodes.length;
                    if (wirisNodeLength > 1) {
                        for (let i = 0; i < wirisNodeLength - 1; i++) {
                            wirisNodes[i].remove();
                            document.getElementsByClassName('wrs_modal_overlay').remove();
                        }
                    }
                }
            }
        };
        // Create an observer instance linked to the callback function		
        let observer = new MutationObserver(callbackOb);
        // Start observing the target node for configured mutations	
        if (targetNode)
            observer.observe(targetNode, config);
}