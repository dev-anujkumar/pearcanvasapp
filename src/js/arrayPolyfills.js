/**
 * This module contains polyfills for Arrays
 */
/* -1- Polyfill for [findLastIndex]
 * finsLastIndex returns the last index at which a given object can be found in the array of objects
 */
Array.prototype && function (ArrayPrototype) {
    ArrayPrototype.findLastIndex = ArrayPrototype.findLastIndex ||
        function (array, cbFunc) {
            if (!array) return -1;
            if (!cbFunc || typeof cbFunc !== "function") throw TypeError(`${cbFunc} is not a function`);
            return array.reduceRight((prev, currentValue, currentIndex) => {
                if (prev > -1) return prev;
                if (cbFunc(currentValue, currentIndex)) return currentIndex;
                return -1;
            }, -1);
        }
}(Array.prototype);