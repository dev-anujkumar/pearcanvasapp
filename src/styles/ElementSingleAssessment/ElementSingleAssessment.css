.figureAssessment{
    font-family: 'Open Sans', sans-serif;
    font-size: 12px;
    max-width:100% !important;
}

.singleAssessmentIdInfo{
    padding-top: 5px!important;
    padding-bottom: 10px!important;
}
.singleAssessmentItemIdInfo{
    padding-bottom: 10px!important;
}
.singleAssessment_Dropdown_Container{
    height: 37px;
}
.singleAssessment_Dropdown_arrow{
float: right;
position: relative;
top: 0px;
right: 8px;
width: 8px;
}
.singleAssessment_Dropdown_activeDropdown{
width: 200px;
float: left;
padding: 8px;
cursor: pointer;
position: relative;
border-radius: 18px;
border: solid 1px silver;
background-color: #ffffff;
}
.singleAssessment_Dropdown_SelectLabel{
    width: 120px;
    float: left;
    font-size: 13px;
    color: #424242;
    position: relative;
    top: 6px; 
}
.singleAssessment_Dropdown_options{
    position: absolute;
    padding-left: 0px;
    margin-left: 122px;
    width: 202px;
    margin-top: 10px;
    max-height: 220px;
    overflow: auto;
    background: #fff;
}
.singleAssessment_Dropdown_item{
    list-style: none;
    padding: 10px;
    background-color: #fff;
    cursor: pointer;
}
.singleAssessment_Dropdown_item:hover{
    background-color: #1565c0;
    color:white;
    
 }
.singleAssessment_Dropdown_currentLabel{
    font-size: 14px;
    margin:0 0 0 8px;
}
.divAssessment{
       max-width:auto!important ;
}

.puf-assessment-id{
    display: none;
}
.figureAssessment .default-img{
    filter: invert(40%) sepia(0%) saturate(0%) hue-rotate(236deg) brightness(110%) contrast(89%);
}

.elm-update-button.embedded-assessment{
    display: inline-block;
    margin: 0px;
}

.elm-status-div.embedded-assessment{
    padding-top: 2px;
    margin: 1px;
}

.singleAssessment_Dropdown_activeDropdown.isElmUpdate{
    margin-right: 25px;
}

.heading4AssessmentItemNumberLabel,.heading4TdxAssessmentItemNumberLabel{
    color: #007FA3;
}

.assessment-metadata {
    display: flex;
    line-height: 17px;
    padding: 9px 0 1px 0;
    vertical-align: middle;
}

div.assessment-metadata-container {
    margin-bottom: 11px;
}

.title-field {
    color: #005a70;
    font-weight: 600;
}

p.single-assessment-title {
    font-weight: 600;
    width: 80px;
}