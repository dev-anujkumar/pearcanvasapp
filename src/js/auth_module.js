import { sendDataToIframe } from '../constants/utility.js'
const configOBJ = require('./../config/config');
let config_object = configOBJ.default;
const IDENTITY_URL = config_object.IDENTITY_URL;
let environment = config_object.NODE_ENV;

/* auth */
require('../auth/openam.js');
import openamConfig from '../auth/openam.js';

/* Session Variables */
let session_token = "";

var BASE_URL;

if ( IDENTITY_URL.indexOf("http") !== 1 ) {
    BASE_URL = IDENTITY_URL;   
} else {
    BASE_URL = "https://"+IDENTITY_URL;
}

var myOpenam;
export const OPEN_AM = myOpenam || {};
