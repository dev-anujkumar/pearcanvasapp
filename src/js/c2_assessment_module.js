const renderderedTagSelector = '#c2-modal';
const configOBJ = require('./../config/config');
let config_object = configOBJ.default;
const CMDS_APIKEY = config_object['CMDS_APIKEY'];
const CMDS_DATA_ENDPOINT = config_object['CMDS_DATA_ENDPOINT'];
const CMDS_SCHEMA_ENDPOINT = config_object['CMDS_SCHEMA_ENDPOINT'];
const CMDS_DATABASE = config_object['CMDS_DATABASE'];
const CMDS_AUTHORIZATION = config_object['CMDS_AUTHORIZATION'];
import { hideTocBlocker, disableHeader } from './toggleLoader'
const WRAPPER_URL = `${config_object.WRAPPER_URL}`;
const authModule = { GET_SSO_TOKEN: function () { return config_object.ssoToken } };
import { sendDataToIframe } from '../constants/utility.js'



// Access individual pattern (from the <script> tag)
var patternBroker; //(PatternBroker && PatternBroker !== undefined && PatternBroker !== null) ? PatternBroker.default : {}; //this.PatternBroker.default;
var patternSearchSelect; //(PatternSearchSelect && PatternSearchSelect !== undefined && PatternSearchSelect !== null) ? PatternSearchSelect.default : {};//this.SearchSCPatterns.default;

let _interactivePattern = {};
let _interactivePatternConfig = {};
var uname = config_object['userId'];

/*Configure the library*/
var libConfig = {
  'locale': 'en_US',
  'headers': {
    'Content-Type': 'application/json',
    'Accept': 'application/ld+json',
    'X-Roles-Test': 'ContentMetadataEditor',
    'Prefer': 'annotation=true',
    'Apikey': CMDS_APIKEY,
    'X-APIKey': CMDS_APIKEY,
    'PearsonSSOSession': authModule.GET_SSO_TOKEN(),
    'X-PearsonSSOSession': authModule.GET_SSO_TOKEN(),
    'Authorization': CMDS_AUTHORIZATION
  },
  'database': CMDS_DATABASE,
  'server': CMDS_DATA_ENDPOINT,
  'taxonomyserver': CMDS_SCHEMA_ENDPOINT,  // Rel 3.6
  'userId': uname || config_object['userId']
};

//patternBroker.setup(libConfig);

export const c2AssessmentModule = {
  searchAndSelectonSave: function (data) {
    console.log("searchAndSelectonSave data: " + '', data);

  },
  launchAssetBrowser: function (fileName, filterType, searchMode, searchSelectAssessmentURN, productId, searchTypeOptVal, callback) {
    patternBroker = PatternBroker.default || {}; //(PatternBroker && PatternBroker !== undefined && PatternBroker !== null) ? PatternBroker.default : {}; //this.PatternBroker.default;
    patternSearchSelect = PatternSearchSelect.default || {};
    disableHeader(true);
    if (_interactivePattern && _interactivePattern.unmount) {
      _interactivePattern.unmount();
    }

    var libConfig = {
      'locale': 'en_US',
      'headers': {
        'Content-Type': 'application/json',
        'Accept': 'application/ld+json',
        'X-Roles-Test': 'ContentMetadataEditor',
        'Prefer': 'annotation=true',
        'Apikey': CMDS_APIKEY,
        'X-APIKey': CMDS_APIKEY,
        'PearsonSSOSession': authModule.GET_SSO_TOKEN(),
        'X-PearsonSSOSession': authModule.GET_SSO_TOKEN(),
        'Authorization': CMDS_AUTHORIZATION
      },
      'database': CMDS_DATABASE,
      'server': CMDS_DATA_ENDPOINT,
      'taxonomyserver': CMDS_SCHEMA_ENDPOINT,  // Rel 3.6
      'searchselectserver': CMDS_DATA_ENDPOINT,
      'userId': uname || config_object['userId']
    };

    patternBroker.setup(libConfig);

    _interactivePatternConfig = { 'selector': renderderedTagSelector };
    // filename = 'Some pre-specified search term - a string value';
    _interactivePatternConfig.filename = fileName;
    // filterType = ['', '', '']  // An array of strings; Each element is a pre-specified type filter; ie.  ["PDF", "Metrodigi Interactive"]
    // _interactivePatternConfig.filterType = ['Journal','Cite interactive: Graph'];

    _interactivePatternConfig.filterType = filterType;

    // searchmode = 'full' OR 'partial'
    //_interactivePatternConfig.searchmode = 'full';
    _interactivePatternConfig.searchmode = searchMode;
    // searchTypeOptVal = 'journal'  OR  'tdx'  OR  'cite'
    //_interactivePatternConfig.searchTypeOptVal = searchTypeOptVal;  // Must be specified as one of the three types
    // searchSelectAssessmentURN - For 2nd-nth question embedding, specify Assessment wURN so the user does not need to search for it again.
    // If specified, the pattern will directly goto screen for selecting underlaying assessment items
    // If it's not specified, the pattern will goto main screen where user will need to research for the assessment
    _interactivePatternConfig.searchSelectAssessmentURN = searchSelectAssessmentURN;
    // productId - Chaucer must capture and supply this value to pattern each time the pattern is used.
    // The pattern will constraint it's search for assessment within specified productId (user does have an option to disable the constraint!)
    // Due to C1 MDS limitation, for now this will be the UUID part of the 'Register' URN. In future it'll be ISBN.
    // On PPE, some sample data exists in : ab1ce445-253a-44a4-a7eb-31274955b2d5
    _interactivePatternConfig.productId = productId;  // Rel 3.6
    // This for INTL lang support. It's still work in progress. Support for 'fr' is forthcoming.
    _interactivePatternConfig.language = 'en';  // Rel 3.6

    _interactivePattern = patternBroker.create('interactivePattern', patternSearchSelect);
    try {
      if (_interactivePattern.corsId) {
        libConfig.headers['Correlation-Id'] = _interactivePattern.corsId;
      }
      patternBroker.setup(libConfig);
      _interactivePattern.setup(_interactivePatternConfig, this.searchAndSelectCallBack);
      _interactivePattern.run(_interactivePattern);
      _interactivePattern.on(callback);

      sendDataToIframe({ 'type': 'hideToc', 'message': {} });

      //alfresco toc issue
      var targetNode = document.getElementsByClassName('overlay-0-0 overlayLittle-0-1')[0];

      // Options for the observer (which mutations to observe)
      var config = { attributes: true };

      // Callback function to execute when mutations are observed
      var callbackOb = function (mutationsList, observercb) {
        for (var mutation of mutationsList) {
          if (mutation.type === 'attributes' && mutation.attributeName === 'class' && targetNode.classList.contains("transitionLeave-0-6")) {
            {
              hideTocBlocker();
              disableHeader(false);
              observercb.disconnect();
            }
          }
        }
      };
      // Create an observer instance linked to the callback function
      var observer = new MutationObserver(callbackOb);

      // Start observing the target node for configured mutations
      observer.observe(targetNode, config);

    } catch (ex1) {
      alert(ex1.message);
      hideTocBlocker();
      disableHeader(false);
    }

  }
}