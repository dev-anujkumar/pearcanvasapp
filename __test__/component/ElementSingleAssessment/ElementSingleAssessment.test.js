import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import { ElementSingleAssessment } from '../../../src/component/ElementSingleAssessment/ElementSingleAssessment';
import { c2AssessmentModule } from '../../../src/js/c2_assessment_module';
import {  singleAssessmentCITEDefault} from '../../../fixtures/ElementSingleAssessmentTestData'
// import config from '../../../src/config/config';
// import { JestEnvironment } from '@jest/environment';

xdescribe('Testing Element Single Assessment component', () => {

    test('renders without crashing', () => {
        const component = mount(<ElementSingleAssessment model={singleAssessmentCITEDefault} index="" />)
        expect(component).toHaveLength(1);
        let instance = component.instance(); 
        expect(instance).toBeDefined();
    })

    let props = {
        model: singleAssessmentCITEDefault,
        index:"1",
        usagetype:"Practice",
        handleFocus: function(){},
        onClick : ()=>{},
        handleBlur: function(){},
     
    };

    let singleAssessment = mount(<ElementSingleAssessment {...props}  />);
    const singleAssessmentInstance = singleAssessment.find('ElementSingleAssessment').instance();
    it('Render Single Assessment default ', () => {
        const spyrenderAssessmentType = jest.spyOn(singleAssessmentInstance, 'renderAssessmentType')
        singleAssessmentInstance.renderAssessmentType(singleAssessmentCITEDefault);
        singleAssessmentInstance.forceUpdate();
        singleAssessment.update();
        expect(spyrenderAssessmentType).toHaveBeenCalledWith(singleAssessmentCITEDefault) 
        spyrenderAssessmentType.mockClear()
    });
    it('onClick Event-Select Assessment usage type', () => {
        singleAssessmentInstance.setState({
            asseessmentUsageTypeDropdown: false,
            activeAsseessmentUsageType: 'Quiz'
        });
        singleAssessmentInstance.forceUpdate();
        singleAssessment.update();
        singleAssessment.find('div.singleAssessment_Dropdown_Container .singleAssessment_Dropdown_activeDropdown').simulate('click');
        singleAssessment.find('ul.singleAssessment_Dropdown_options>li:first-child').simulate('click');
        singleAssessment.find('div.pearson-component.image').simulate('click');
        singleAssessment.find('div.figureElement').simulate('click');
        expect(singleAssessmentInstance.state.showAssessmentPopup).toBe(true);
        expect(singleAssessmentInstance.state.asseessmentUsageTypeDropdown).toBe(false);
        expect(singleAssessmentInstance.state.activeAsseessmentUsageType).toBe('Quiz');
           
    });
    it('Test-handleAssessmentTypeChange function', () => {
        singleAssessmentInstance.setState({
            showAssessmentPopup: true
        });
        singleAssessmentInstance.forceUpdate();
        const spyhandleAssessmentTypeChange = jest.spyOn(singleAssessmentInstance, 'handleAssessmentTypeChange')
        singleAssessmentInstance.handleAssessmentTypeChange('Practice');
        singleAssessmentInstance.forceUpdate();
        singleAssessment.update();
        expect(spyhandleAssessmentTypeChange).toHaveBeenCalledWith('Practice') 
        expect(singleAssessmentInstance.state.showAssessmentPopup).toBe(true);
        expect(singleAssessmentInstance.state.asseessmentUsageTypeDropdown).toBe(false);
        spyhandleAssessmentTypeChange.mockClear()
    });
    it('Test-toggleAssessmentPopup function', () => {
        singleAssessmentInstance.setState({
            showAssessmentPopup: true
        });
        singleAssessmentInstance.forceUpdate();
        const spytoggleAssessmentPopup = jest.spyOn(singleAssessmentInstance, 'toggleAssessmentPopup')
        singleAssessmentInstance.toggleAssessmentPopup(false);
        singleAssessmentInstance.forceUpdate();
        singleAssessment.update();
        expect(spytoggleAssessmentPopup).toHaveBeenCalledWith(false) 
        expect(singleAssessmentInstance.state.showAssessmentPopup).toBe(false)
        spytoggleAssessmentPopup.mockClear()
    });
    it('Test-toggleUsageTypeDropdown function', () => {
        singleAssessmentInstance.setState({
            asseessmentUsageTypeDropdown: true
        });
        singleAssessmentInstance.forceUpdate();
        const spytoggleUsageTypeDropdown = jest.spyOn(singleAssessmentInstance, 'toggleUsageTypeDropdown')
        singleAssessmentInstance.toggleUsageTypeDropdown(true);
        singleAssessmentInstance.forceUpdate();
        singleAssessment.update();
        expect(spytoggleUsageTypeDropdown).toHaveBeenCalledWith(true) 
        expect(singleAssessmentInstance.state.asseessmentUsageTypeDropdown).toBe(false)
        spytoggleUsageTypeDropdown.mockClear()
    });
    it('Test-handleAssessmentFocus function', () => {
        const spyhandleAssessmentFocus = jest.spyOn(singleAssessmentInstance, 'handleAssessmentFocus')
        singleAssessmentInstance.handleAssessmentFocus();
        singleAssessmentInstance.forceUpdate();
        singleAssessment.update();
        expect(spyhandleAssessmentFocus).toHaveBeenCalled()
        spyhandleAssessmentFocus.mockClear()
    });
    it('Test-handleAssessmentBlur function', () => {
        const spyhandleAssessmentBlur = jest.spyOn(singleAssessmentInstance, 'handleAssessmentBlur')
        singleAssessmentInstance.handleAssessmentBlur();
        singleAssessmentInstance.forceUpdate();
        singleAssessment.update();
        expect(spyhandleAssessmentBlur).toHaveBeenCalled()
        spyhandleAssessmentBlur.mockClear()
    });
    it('onClick C2 media', () => {
        let props = {
            handleFocus: function(){},
            handleBlur : jest.fn(),
            model : singleAssessmentCITEDefault,
            onClick : ()=>{},
        }
        let assessmentData={
            id: "urn:pearson:work:133dd9fd-a5be-45e5-8d83-891283abb9a5",
            itemID: "urn:pearson:work:fb9bcb66-3073-45e6-ab8a-b595a35bf93b",
            title: "Open response question updated",
            itemsData: {taxonomicType:["cite"]}

        }
        const singleAssessment = mount(<ElementSingleAssessment {...props}/>);
        const singleAssessmentInstance = singleAssessment.find('ElementSingleAssessment').instance();
        singleAssessmentInstance.launchAssetBrowserCallBack(assessmentData);
    })
    it('Test-C2 media handling',()=>{
        let props = {
            handleFocus: function(){},
            handleBlur : jest.fn(),
            model : singleAssessmentCITEDefault,
            onClick : ()=>{},
        }
        const singleAssessment = mount(<ElementSingleAssessment {...props}/>);
        const singleAssessmentInstance = singleAssessment.find('ElementSingleAssessment').instance();
        const spyhandleC2AssessmentClick = jest.spyOn(singleAssessmentInstance, 'handleC2AssessmentClick')
        singleAssessmentInstance.handleC2AssessmentClick("");
        c2AssessmentModule.launchAssetBrowser('','','','','','',()=>{});
        singleAssessmentInstance.forceUpdate();
        singleAssessment.update();
        expect(spyhandleC2AssessmentClick).toHaveBeenCalled()
        spyhandleC2AssessmentClick.mockClear() 
    })
    describe('Test- launchAssetBrowserCallBack function',()=>{
        let props = {
            handleFocus: function(){},
            handleBlur : jest.fn(),
            model : singleAssessmentCITEDefault,
            onClick : ()=>{},
        }
        const singleAssessment = mount(<ElementSingleAssessment {...props}/>);
        const singleAssessmentInstance = singleAssessment.find('ElementSingleAssessment').instance();
        it('Test- launchAssetBrowserCallBack if-case', () => {
            let assessmentData = {
                id: "urn:pearson:work:133dd9fd-a5be-45e5-8d83-891283abb9a5",
                itemID: "urn:pearson:work:fb9bcb66-3073-45e6-ab8a-b595a35bf93b",
                title: "Open response question updated",
                itemsData: { taxonomicType: ["cite"] }
            }
            const spylaunchAssetBrowserCallBack = jest.spyOn(singleAssessmentInstance, 'launchAssetBrowserCallBack')
            singleAssessmentInstance.launchAssetBrowserCallBack(assessmentData);
            singleAssessmentInstance.forceUpdate();
            singleAssessment.update();
            expect(spylaunchAssetBrowserCallBack).toHaveBeenCalled()
            spylaunchAssetBrowserCallBack.mockClear()
        })
        it('Test- launchAssetBrowserCallBack if-else-case', () => {
            let assessmentData = {
                id: "urn:pearson:work:133dd9fd-a5be-45e5-8d83-891283abb9a5",
                itemID: "urn:pearson:work:fb9bcb66-3073-45e6-ab8a-b595a35bf93b",
                title: "Open response question updated",
                itemsData: {}
            }
            const spylaunchAssetBrowserCallBack = jest.spyOn(singleAssessmentInstance, 'launchAssetBrowserCallBack')
            singleAssessmentInstance.launchAssetBrowserCallBack(assessmentData);
            singleAssessmentInstance.forceUpdate();
            singleAssessment.update();
            expect(spylaunchAssetBrowserCallBack).toHaveBeenCalled()
            spylaunchAssetBrowserCallBack.mockClear()
        })
        it('Test- launchAssetBrowserCallBack if-else-case', () => {
            let assessmentData = {
                id: "urn:pearson:work:133dd9fd-a5be-45e5-8d83-891283abb9a5",
                itemID: "urn:pearson:work:fb9bcb66-3073-45e6-ab8a-b595a35bf93b",
                title: "Open response question updated",
                assessmentData: { taxonomicType: ["cite"] }
            }
            const spylaunchAssetBrowserCallBack = jest.spyOn(singleAssessmentInstance, 'launchAssetBrowserCallBack')
            singleAssessmentInstance.launchAssetBrowserCallBack(assessmentData);
            singleAssessmentInstance.forceUpdate();
            singleAssessment.update();
            expect(spylaunchAssetBrowserCallBack).toHaveBeenCalled()
            spylaunchAssetBrowserCallBack.mockClear()
        })
    })
});