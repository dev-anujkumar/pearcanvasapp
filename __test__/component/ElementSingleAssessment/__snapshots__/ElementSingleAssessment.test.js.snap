// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Element Single Assessment component Render Single Assessment default  1`] = `
<ElementSingleAssessment
  elementId={[Function]}
  index={[Function]}
  model={
    Object {
      "captions": Object {
        "footnotes": Array [],
        "mathml": Array [],
        "schema": "http://schemas.pearson.com/wip-authoring/authoredtext/1#/definitions/authoredtext",
        "text": "",
        "textsemantics": Array [],
      },
      "contentUrn": "urn:pearson:entity:0f7c431f-e6c9-4574-bfe0-7d4ac25c1032",
      "credits": Object {
        "footnotes": Array [],
        "mathml": Array [],
        "schema": "http://schemas.pearson.com/wip-authoring/authoredtext/1#/definitions/authoredtext",
        "text": "",
        "textsemantics": Array [],
      },
      "figuredata": Object {
        "elementdata": Object {
          "assessmentformat": "cite",
          "assessmentid": "urn:pearson:work:133dd9fd-a5be-45e5-8d83-891283abb9a5",
          "assessmentitemid": "urn:pearson:work:fb9bcb66-3073-45e6-ab8a-b595a35bf93b",
          "assessmentitemtype": "assessmentItem",
          "posterimage": Object {
            "path": "https://cite-media-stg.pearson.com/legacy_paths/8efb9941-4ed3-44a3-8310-1106d3715c3e/FPO-assessment.png",
          },
          "schema": "http://schemas.pearson.com/wip-authoring/assessment/1#/definitions/assessment",
          "usagetype": "Diagnostic",
        },
        "schema": "http://schemas.pearson.com/wip-authoring/element/1",
        "subtype": "assessment",
        "type": "element-assessment",
      },
      "figuretype": "assessment",
      "id": "urn:pearson:work:47926695-265e-469e-bfc3-2c942d2c1824",
      "schema": "http://schemas.pearson.com/wip-authoring/figure/1",
      "subtitle": Object {
        "footnotes": Array [],
        "mathml": Array [],
        "schema": "http://schemas.pearson.com/wip-authoring/authoredtext/1#/definitions/authoredtext",
        "text": "",
        "textsemantics": Array [],
      },
      "subtype": "",
      "title": Object {
        "schema": "http://schemas.pearson.com/wip-authoring/authoredtext/1#/definitions/authoredtext",
        "text": "Open response question updated",
      },
      "type": "figure",
      "versionUrn": "urn:pearson:work:47926695-265e-469e-bfc3-2c942d2c1824",
    }
  }
>
  <div
    className="figureElement"
  >
    <div
      className="divAssessment"
    >
      <figure
        className="figureAssessment"
      >
        <header>
          <h4
            className="heading4ImageTextWidthNumberLabel"
          >
            Open response question updated
          </h4>
        </header>
        <div
          className="singleAssessmentIdInfo"
        >
          <strong>
            ID: 
          </strong>
          urn:pearson:work:133dd9fd-a5be-45e5-8d83-891283abb9a5
        </div>
        <div
          className="singleAssessmentItemIdInfo"
        >
          <strong>
            ITEM ID: 
          </strong>
          urn:pearson:work:fb9bcb66-3073-45e6-ab8a-b595a35bf93b
        </div>
        <div
          className="singleAssessment_Dropdown_Container"
        >
          <div
            className="singleAssessment_Dropdown_SelectLabel"
          >
            Select usage type
          </div>
          <div
            className="singleAssessment_Dropdown_activeDropdown notselect"
            onClick={[Function]}
          >
            <span
              className="singleAssessment_Dropdown_currentLabel"
            >
              Quiz
            </span>
            <span
              className="singleAssessment_Dropdown_arrow"
            >
              <svg
                className="dropdown-arrow"
                viewBox="0 0 9 4.5"
              >
                <path
                  d="M0,0,4.5,4.5,9,0Z"
                />
              </svg>
            </span>
          </div>
        </div>
        <div
          className="pearson-component image"
          data-type="image"
          data-uri=""
        >
          <img
            alt=""
            class="imageTextWidth lazyloaded imageeee"
            data-src="https://cite-media-stg.pearson.com/legacy_paths/8efb9941-4ed3-44a3-8310-1106d3715c3e/FPO-assessment.png"
            src="https://cite-media-stg.pearson.com/legacy_paths/8efb9941-4ed3-44a3-8310-1106d3715c3e/FPO-assessment.png"
            title="View Image"
          />
        </div>
      </figure>
    </div>
  </div>
</ElementSingleAssessment>
`;
